src/components/UserUpdateScreen.js

```diff
@@ -0,0 +1,152 @@
+// @flow
+import React, {useLayoutEffect, useState, useCallback} from 'react';
+import type {Node} from 'react';
+import {View, Text, Button, StyleSheet} from 'react-native';
+import type {RouteProp} from '@react-navigation/native';
+import type {StackNavigationProp} from '@react-navigation/stack';
+import {useDispatch} from 'react-redux';
+import {addUser} from '../UsersAction';
+import _ from 'lodash';
+import type {UserDispatch, User} from '../types.js';
+import UserInput from './UserInput';
+import DateTimePicker from '@react-native-community/datetimepicker';
+
+type UserUpdateScreenProps = {
+  navigation: StackNavigationProp<any, any>,
+  route: RouteProp<any, any>,
+};
+
+const styles = StyleSheet.create({
+  detailsContainer: {
+    margin: 10,
+  },
+  locationLabel: {
+    fontSize: 18,
+    marginVertical: 10,
+  },
+  locationContainer: {
+    marginLeft: 10,
+  },
+  dobContainer: {
+    flexDirection: 'row',
+    alignItems: 'center',
+  },
+  dobLabel: {
+    flexBasis: '25%',
+    fontSize: 16,
+    marginVertical: 3,
+  },
+  dobValue: {
+    flexBasis: '75%',
+    marginVertical: 3,
+  },
+});
+
+const UserUpdateScreen = ({navigation, route}: UserUpdateScreenProps): Node => {
+  const dispatch = useDispatch<UserDispatch>();
+  const newUser = {
+    name: {
+      first: '',
+      last: '',
+    },
+    email: '',
+    phone: '',
+    cell: '',
+    gender: '',
+  };
+  const [user, setUser] = useState<User>(newUser);
+
+  const writeValue = (value: string | number, path: string): void => {
```

Comment:
Could we use `useCallback` in this case

Reply:
I suppose that we can't since user is a deep object and the reference is actually changed every time we call this function. Hence, there is no point in using it.